### ISSUE TITLE
Columns incorrect resizing when scrolled to far right

### ISSUE THREAD

**Issue by KoromaruKoruko** [dotnet/wpf#8383]
### Description

If you scroll the horizontal scrollbar to the far right on a datagrid then attempt to resize a column it will produce unpredictable and incorrect resizing.

### Reproduction Steps

Create datagrid.
Have columns go further the the view.
Scroll to the very far right.
Try and resize a column.

### Expected behavior

Resizing will follow the cursor and break the scroll docking if necessary.

### Actual behavior

Resizing will continue to have the scrollbar locked to the far right and cause the column to sporadically change size while moving the cursor.

### Regression?

Unknown

### Known Workarounds

Unknown

### Impact

Poor user experience when faced with this particular input.

### Configuration

X64 .net6-windows

**Comment by singhashish-wpf** [comment]
@KoromaruKoruko Could you please share your minimal repro which you are testing?

**Comment by KoromaruKoruko** [comment]
I do not have one as this was discovered while working on business code, I'll throw one together over the weekend if I have time.

**Comment by gekka** [comment]
@KoromaruKoruko,   I think it's same issue as [this link](https://learn.microsoft.com/en-us/answers/questions/102861/wpf-datagrid-resizing-column-causes-extreme-scroll).

1. Start dragging thumb in Column Header.
2. Move the mouse.
3. ColumnWidth changes because distance changed.
4. Changing the column width causes horizontal scrolling in the DataGrid.
5. The scrolling changes screen position of the thumb, and the changes distance between the thumb and the mouse.
6. repeat from 3

![20231109](https://github.com/dotnet/wpf/assets/10055381/24eb7853-f3ae-41e5-b818-e2bc2056d277)

**Comment by KoromaruKoruko** [comment]
> ![20231109](https://user-images.githubusercontent.com/10055381/281673295-24eb7853-f3ae-41e5-b818-e2bc2056d277.gif) [ ![20231109](https://user-images.githubusercontent.com/10055381/281673295-24eb7853-f3ae-41e5-b818-e2bc2056d277.gif) ](https://user-images.githubusercontent.com/10055381/281673295-24eb7853-f3ae-41e5-b818-e2bc2056d277.gif) [ ](https://user-images.githubusercontent.com/10055381/281673295-24eb7853-f3ae-41e5-b818-e2bc2056d277.gif)

yes, if you try any of the columns other then the very last one, the problem is much much much worse.

**Comment by cbra-caa** [comment]
@singhashish-wpf 

Here with a gif showing the problem, and the attached project from which the gif is created as it is still a problem.

Also, I would like to draw attention to the solution proposed on the linked issue: https://gist.github.com/gekka/4b85228ac7ad29457145db0978f9cfd4 which makes the scroll experience a lot more pleasant.
Currently running with it as an attached behavior, which simplified the hookin quite a bit.

![Image](https://github.com/user-attachments/assets/b6a7ae3f-7100-4a8c-be8d-094b4c96cfd3)

[DataGridResizeColumnBug.zip](https://github.com/user-attachments/files/21163490/DataGridResizeColumnBug.zip)
